library/spdm_requester_lib/libspdm_req_vendor_request.c

```diff
@@ -175,12 +210,28 @@ libspdm_return_t libspdm_try_vendor_send_request_receive_response(
         status = LIBSPDM_STATUS_INVALID_MSG_SIZE;
         goto done;
     }
-    /* check response buffer size at least spdm response default header plus
-     * number of bytes required by vendor id and 2 bytes for response payload size */
-    if (spdm_response_size < sizeof(spdm_vendor_defined_response_msg_t) +
-        spdm_response->vendor_id_len + sizeof(uint16_t)) {
-        status = LIBSPDM_STATUS_INVALID_MSG_SIZE;
-        goto done;
+
+    if (use_large_payload) {
```

Comment:
Bit[7] of `Param1` in the response must equal Bit[7] of `Param1` in the request.